File: app/src/main/java/de/trbnb/databindingcommandsapp/MainViewModel.java
package de.trbnb.databindingcommandsapp;

import android.content.Context;
import android.databinding.BaseObservable;
import android.widget.Toast;

import de.trbnb.databindingcommands.command.SimpleCommand;
import de.trbnb.databindingcommands.functions.Action;

public class MainViewModel extends BaseObservable {

    private SimpleCommand buttonCommand = new SimpleCommand(new Action<Context>() {
        @Override
        public void invoke(Context context) {
            Toast.makeText(context, "It just works!", Toast.LENGTH_LONG).show();
        }
    }, false);

    public SimpleCommand getButtonCommand() {
        return buttonCommand;
    }
}

File: app/src/main/java/de/trbnb/databindingcommandsapp/MainViewModel.kt
package de.trbnb.databindingcommandsapp

import android.content.Context
import android.databinding.BaseObservable
import android.widget.Toast
import de.trbnb.databindingcommands.command.SimpleCommand

class MainViewModel(private val appContext: Context) : BaseObservable() {

    val buttonCommand = SimpleCommand {
        Toast.makeText(appContext, "It just works!", Toast.LENGTH_LONG).show()
    }

    val longClickCommand = SimpleCommand {
        Toast.makeText(appContext, "This is long click!", Toast.LENGTH_LONG).show()
        true
    }
}
